src/net.h

```diff
@@ -798,11 +798,11 @@ class CConnman
     // alias for thread safety annotations only, not defined
     RecursiveMutex& GetNodesMutex() const LOCK_RETURNED(m_nodes_mutex);
 
-    bool ForNode(NodeId id, std::function<bool(CNode* pnode)> func);
+    bool ForNode(NodeId id, std::function<bool(CNodeRef pnode)> func);
```

Comment:
I don't think this makes sense in general? If you have a shared_ptr, you should normally just pass the reference to the object (`CNode&`) if you've checked it's not null, or the pointer itself (`CNode*`).

cf http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-smart